File: DSA_Codesheet/Python/Selection_Sort.py
def selection_sort(arr):

    """
    Function for implementation of Selection Sort using python.
    """

    n = len(arr)
    
    for i in range(n):
        # Find the minimum element in the unsorted portion of the array
        min_index = i
        for j in range(i + 1, n):
            if arr[j] < arr[min_index]:
                min_index = j
        
        # Swap the minimum element with the first element in the unsorted portion
        arr[i], arr[min_index] = arr[min_index], arr[i]

if __name__ == "__main__":
    try:
        arr = list(map(int, input("Enter values for array: ").split())) 
    except:
        print("Please enter a valid number")
    print(f"Your original array: {arr}")
    selection_sort(arr)
    print(f"Sorted Array: {arr}")

File: DSA_Codesheet/Java/SelectionSort.java
import java.util.Scanner;

public class SelectionSort {
    public static void selectionSort(int[] arr) {
        int n = arr.length;

        for (int i = 0; i < n - 1; i++) {
            // Find the minimum element in the unsorted portion of the array
            int minIndex = i;
            for (int j = i + 1; j < n; j++) {
                if (arr[j] < arr[minIndex]) {
                    minIndex = j;
                }
            }

            // Swap the found minimum element with the element at index i
            int temp = arr[i];
            arr[i] = arr[minIndex];
            arr[minIndex] = temp;
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        int n = scanner.nextInt();
        
        int[] arr = new int[n];
        
        for (int i = 0; i < n; i++) {
            arr[i] = scanner.nextInt();
        }
        
        selectionSort(arr);
        
        System.out.println("Sorted array:");
        for (int value : arr) {
            System.out.print(value + " ");
        }
        
        // Close the scanner to free up resources
        scanner.close();
    }
}
